gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/handler/authorization/AuthorizationRequestFailureHandler.java

```diff
@@ -296,14 +297,15 @@ private boolean isForbiddenRedirectUri(Client client, AuthorizationRequest autho
     }
 
     /**
-     * Application, only with 'client_credentials' grant type cannot have
+     * Application, only with 'client_credentials' or 'urn:ietf:params:oauth:grant-type:jwt-bearer' grant type cannot have
      * `redirect_uri` in the request
      */
     private boolean invalidParamRedirectURI(Client client, String redirectURI) {
         return client != null
                 && client.getAuthorizedGrantTypes() != null
-                && client.getAuthorizedGrantTypes().size() == 1
-                && client.getAuthorizedGrantTypes().contains(CLIENT_CREDENTIALS)
-                && redirectURI != null;
+                && client.getAuthorizedGrantTypes().stream()
+                    .filter(grantType -> !CLIENT_CREDENTIALS.equals(grantType) || !JWT_BEARER.equals(grantType))
```

Comment:
is it always true?
for grantType==CLIENT_CREDENTIALS it evaluates to ~TRUE OR ~FALSE => FALSE OR TRUE => TRUE
for grantType==JWT_BEARER it evaluates to ~FALSE OR ~TRUE => TRUE OR FALSE => TRUE
for grantType==XXX ~FALSE OR ~FALSE => TRUE OR TRUE = TRUE
?